xdsl/dialects/builtin.py

```diff
@@ -141,6 +141,12 @@ def from_string_attr(root: StringAttr,
                          nested: List[StringAttr] = []) -> SymbolRefAttr:
         return SymbolRefAttr([root, ArrayAttr(nested)])
 
+    def string_value(self):
```

Comment:
Feels like you could do this with a `join` to make it more pythonic.

Reply:
Well, that's more or less one list comprehension, right? I am fine with the version as is as well :)